site-src/geps/gep-1282.md

```diff
@@ -98,12 +106,292 @@ Of course, from the section above, you can see that there are many other feature
+// 
+type BackendTargetReference struct {
+    // Group is the group of the target resource.
+    Group Group `json:"group"`
+
+    // Kind is kind of the target resource.
+    // Valid values include:
+    // * "Service" (the default if no Kind is specified)
+    // * "ServiceImport"
+    //
+    // Invalid values include:
+    //
+    // * "invalid/kind" - "/" is an invalid character
+    // +required
+    Kind string `json:"kind"`
+
+    // Name is the name of the target resource.
+    Name ObjectName `json:"name"`
+}
+
+// BackendCapabilitiesSpec defines a group of specific properties held by its
+// affiliated backend, and specified by the backend owner (typically the 
+// application developer).  
+//
+// Examples of properties that belong in the BackendCapabilitiesSpec include:
+// TLS re-encryption, session state, timeouts, retries, Websockets, HTTP/2, 
+// load balancer algorithm, healthcheck info, external traffic policy,
+// connection limits, error page, protocol switch, alerts, and others. 
+//
+// Support: Extended
+//
+type BackendCapabilitiesSpec struct {
+    // Backend refers to the object to which we apply these capabilities
+	// +required
+	Backend backendTargetReference `json:"backend,omitempty"`
+	
+    // TLS is the section of details on encryption processing in the traffic 
+    // hop from Gateway to backend, sometimes known as upstream TLS or re-encrypt.
+    // A xRoute configured as passthrough cannot specify a TLS encrypt type.
+    // TLS is useful for HTTPRoute and should not be used on UDP/TCP/TLSRoute.
+    //
+    // +optional
+    TLS *EncryptType `json:"tls,omitempty"`
+}
+
+// EncryptType describes the details needed by the client (Gateway) to implement 
+// encryption from the client to the destination endpoint.
+type EncryptType struct {
+    // The minimal TLSVersion that is accepted by the endpoint.
+    // If empty, there is no minimal backend TLSVersion.
+    // +optional	
+    MinVersion   *TLSVersion `json:"minVersion,omitempty"`
+
+    // The maximum TLSVersion that is accepted by the endpoint.
+    // If empty, there is no maximum backend TLSVersion.
+    // +optional    
+    MaxVersion   *TLSVersion  `json:"maxVersion,omitempty"`
+    
+    // CipherSuites is the minimum cipher suite that the backend accepts.
+    // If empty, anything that is appropriate for the MinVersion is accepted.
+    // If the MinVersion is not specified, the result is implementation
+    // dependent, e.g. based on the minimimal TLS version supported by the
+    // implementation.
+    // +optional
+    CipherSuites []string `json:"cipherSuites,omitempty"`
+    
+    // CACerts is the PEM bundle containing Certificate Authority, trusted by both 
+    // client and endpoint, and intermediate certificates needed to complete the 
+    // certificate chain of trust.  If empty, the system root certs should be used.
+    //
+    // +optional	
+    CACerts      *SecretObjectReference `json:"caCerts,omitempty"`
```

Comment:
That's the biggest problem - we can't have a SecretObjectReference crossing a namespace boundary. We had CVEs related to this, it's not good. 

Also not sure what it means 'trusted by both' - I suspect what you want to define here is the root CA in case the backend is using a private root. SPKI or the public key or cert - inline, like it is used in kubeconfig - may work.
However I don't know if this API is the proper place to define root key managed and distribution - there is a working 
group I think working on certs and security. 

I don't disagree that supporting serlf-signed or private root is important - just that PKI config is probably better as a separate concern, not part of backend properties - and can't be based on cross namespace Secret references ( or expectation that each namespace has a secret with that name).

Reply:
This Secret reference is not for some other service to connect to this one, it's for the Gateway controller (which already needs cluster-wide access to secrets to be able to do its job). The main, relevant client here is the Gateway controller. 

And even if users _do_ need to do a cross-namespace Secret reference, that's exactly what ReferenceGrant is for, to enable the _controlled_ use of Secrets or other resources across namespaces. It seems reasonable to assume that, for self-signed certs, a CA cert might live in its own namespace, with a ReferenceGrant then being required to allow access to it. Or, of course, a controller could just copy that into relevant namespaces too.

Lastly, this is a CA _certificate chain_, not a _key_, so it's intended to be public. We could use another object (maybe a configmap?) to store it if necessare.